pkg/otlp/metrics/metrics_translator.go

```diff
@@ -216,7 +216,7 @@ func (t *Translator) mapNumberMonotonicMetrics(
 
 		if dx, ok := t.prevPts.MonotonicDiff(pointDims, startTs, ts, val); ok {
 			consumer.ConsumeTimeSeries(ctx, pointDims, Count, ts, dx)
-		} else if i == 0 && t.shouldConsumeInitialValue(startTs, ts) {
+		} else if i == 0 && dx >= 0 && t.shouldConsumeInitialValue(startTs, ts) {
```

Comment:
Not sure I get this, if `dx == 0`, then we know the point in the cache is more recent, so maybe we should not report this first point?

Reply:
If there is no point for the metric in the cache, [this code](https://github.com/DataDog/opentelemetry-mapping-go/blob/8a7737db71b42ace42d41b490fe158e7e168a799/pkg/otlp/metrics/ttlcache.go#L77-L88) won't run and `dx` will be zero. In this case, we need to compute the first point as this point not being in the cache indicate this is likely the first value.

I changed the return [here](https://github.com/DataDog/opentelemetry-mapping-go/pull/228/files#diff-0fb69cae87533a77406de810cfa4b0233f043eb777668b55f43cf46623796325R82) to `-1` so we know that the value is not the first point if this is negative.